src/org/labkey/cds/view/template/FrontPage.jsp

```diff
@@ -73,6 +80,69 @@
                 $('#notification').remove();
             });
         });
+
+        LABKEY.Ajax.request({
+            url : LABKEY.ActionURL.buildURL('cds', 'news.api'),
+            success: LABKEY.Utils.getCallbackWrapper(function(response) {
+                if (response.success)
+                {
+                    if (response.items) {
+                        var items = response.items;
+
+                        var options = { year: 'numeric', month: 'long', day: 'numeric' };
+
+                        var $table = $('<table>').appendTo($('#recentBlogPosts'));
+
+                        var $tr = $('<tr class="thumbnail">').appendTo($table);
+
+                        var $displayThumbnail = function(imageIdx) {
+                            var canvas = document.createElement("canvas");
+                            canvas.width = 176.761;
+                            canvas.height = 100;
+                            canvas.style.border = "1px solid #DCDCDC";
+                            var c = canvas.getContext("2d");
+
+                            var img = new Image();
+                            img.src = "<%=h(baseURL + blogPath)%>" + items[imageIdx].imageFile;
+
+                            console.log("img src = " + img.src);
+                            img.onload = function () {
+                                c.drawImage(img, 0, 0, canvas.width, canvas.height);
+                            }
+                            document.getElementById("thumbnail" + (imageIdx+1)).appendChild(canvas);
+                        };
+
+                        $('<td>').html('<a id="thumbnail1" href="' + items[0].link + '"></a>').appendTo($tr);
```

Comment:
Could we have a value for the max number of thumbnails to render and just loop over the news feed items to create the elements? It looks like this code will blow up if less than 4 items are returned from the news feed. 

Do we need to be more fault tolerant on whether all item properties may be present? Or is this validation happening elsewhere?

Reply:
Now iterating through max number of items.

Did not make it fault tolerant (or initially add any checks) since these blog items are already in use to display on the Dataspace's home page, and there already are more than 4 blogs present, and new ones will always get added. And if the image value is not present then it will show an empty box like so:
![image](https://github.com/LabKey/cds/assets/10765266/dfe935db-3e61-46e7-ab13-ff8dedc23ce2)